src/main/java/me/coley/recaf/mapping/TSrgMappings.java

```diff
@@ -0,0 +1,66 @@
+package me.coley.recaf.mapping;
+
+import me.coley.recaf.util.StringUtil;
+import me.coley.recaf.workspace.Workspace;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * TSRG mappings file implementation.
+ *
+ * @author Filip (1fxe)
+ */
+public class TSrgMappings extends FileMappings {
+    private static final String FAIL = "Invalid TSrg mappings, ";
+
+    /**
+     * Constructs mappings from a given file.
+     *
+     * @param path      A path to a file containing TSrg styled mappings.
+     * @param workspace Workspace to pull names from when using hierarchy lookups.
+     * @throws IOException Thrown if the file could not be read.
+     */
+    TSrgMappings(Path path, Workspace workspace) throws IOException {
+        super(path, workspace);
+    }
+
+    @Override
+    protected Map<String, String> parse(String text) {
+        Map<String, String> map = new HashMap<>();
```

Comment:
There is nearly always one map entry per line of text. Perhaps it would be good to initialise the HashMap after the `StringUtil.splitNewline` and pass it `lines.length`?